pkg/fileutil/io.go

```diff
@@ -179,11 +179,19 @@ func VerifySafeFilename(absPath string) error {
 	if !filepath.IsAbs(absPath) {
 		return fmt.Errorf("relative path not allowed: %w", ErrInvalidPath)
 	}
-	filename, err := filepath.EvalSymlinks(absPath)
-	if err != nil {
+	if err := EvalSymlink(absPath); err != nil {
```

Comment:
This can be messy and slow. Also the implementation may miss the case of a dumping to itself - depending on how cyclic symlink expansion is handled. 
Prefer to use the Symlink bit on [FileMode returned by Stat](https://pkg.go.dev/io/fs#FileMode). It is faster, precise, and does not depend on precisely understanding the implementation.

Reply:
Added tests + modified configuration param and check for all irregular files